Civilian Hud

SingingSpock opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

Adds a new form of HUD to the game, the Civilian HUD, and its related special types. The Civilian HUD lets you see what job is displayed on the ID of those around you and nothing else, no arrest status or implants or anything. Gives the captain and HoP the fancy new civilian HUDsunglasses to replace the ones that start on their face and in their wardrobe respectively. These items can be printed from any department lathe, with the exception of the implant version which can only be printed from the mechfab and medical lathe as per usual for implants. The sprites are shamelessly stolen from Paradise, but I was unable to figure out how to set the worn sprite, if someone could tell me how so people aren't walking around with pink-and-black blobs on their eyes I'd appreciate it.

## Why It's Good For The Game

The ability to see someone's job at a glance without having to shift-click someone who may be sprinting across the screen is useful for more than just the people who have Security HUDs on all the time. It doesn't give players access to information they wouldn't otherwise have, just makes it far more readily visible. 

## Changelog

<!-- If your PR modifies aspects of the game that can be concretely observed by players or admins you should add a changelog. If your change does NOT meet this description, remove this section. Be sure to properly mark your PRs to prevent unnecessary GBP loss. You can read up on GBP and it's effects on PRs in the tgstation guides for contributors. Please note that maintainers freely reserve the right to remove and add tags should they deem it appropriate. You can attempt to finagle the system all you want, but it's best to shoot for clear communication right off the bat. -->

:cl:
add: Civilian HUDs that only let you see jobs
add: Civilian HUDsunglasses to the Captain and HoP
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SingingSpock committed:
Civilian Hud

Mothblocks commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1229,6 +1231,7 @@
 		"ci-diaghud",
 		"ci-gloweyes",
 		"ci-medhud",
+		"ci-civhud",
```
ABC order

Mothblocks commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1145,6 +1146,7 @@
 		"nvgmesons",
 		"nv_scigoggles",
 		"security_hud_night",
+		"civ_hud_night",
```
ABC order

Mothblocks commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1130,6 +1130,7 @@
 		"health_hud",
 		"scigoggles",
 		"security_hud",
+		"civ_hud",
```
ABC order

SingingSpock committed:
abc order

Fikou commented on code/modules/clothing/glasses/hud.dm:
```diff
@@ -55,6 +55,30 @@
+	hud_type = DATA_HUD_SECURITY_BASIC
+	hud_trait = TRAIT_CIVILIAN_HUD
+
+/obj/item/clothing/glasses/hud/civ/night
```
whats the point of this? i can just print an nvg sechud

SingingSpock replied:
This is printable at all fabs, not just the security fab. Additionally, some security players (especially downstream on Skyrat) will have an issue with you running around with a secHUD.

ShizCalev replied:
i agree, good for downstream.

Fikou commented on code/modules/clothing/glasses/hud.dm:
```diff
@@ -55,6 +55,30 @@
+	name = "civilian HUD"
+	desc = "A heads-up display that shows only the current job ID of the user."
+	icon_state = "civ"
+	hud_type = DATA_HUD_SECURITY_BASIC
```
wouldnt you want to create your own hud type in the case the basic security huds get another thing?

SingingSpock replied:
Basic security HUDs use the SECURITY_ADVANCED hud_type as well. That is the HUD layer that gives you all the security information. Nothing in the code currently uses SECURITY_BASIC, which is already designed to do exactly what I intend it to do. Adding a copy/paste of the code on the off-chance that this gets drastically changed in the future when it hasn't been touched in forever is bad spaghetti

Fikou commented on code/modules/surgery/organs/augments_eyes.dm:
```diff
@@ -31,6 +31,12 @@
 		REMOVE_TRAIT(eye_owner, HUD_trait, ORGAN_TRAIT)
 	..()
 
+/obj/item/organ/internal/cyberimp/eyes/hud/civ
```
with this one too, i can just get a sechud implant

SingingSpock replied:
And again, some security players will have an issue with it. Additionally, I have plans to make this available as a loadout item downstream.

ShizCalev replied:
i agree, good for downstream.

Fikou commented on code/modules/research/designs/misc_designs.dm:
```diff
@@ -3,6 +3,24 @@
 
+/datum/design/civ_hud
+	name = "Civilian HUD"
+	desc = "A heads-up display that shows only the current job ID of the user."
```
this description is misleading and also worded really weirdly

Fikou replied:
the other ones are too tbh (the items as well)

SingingSpock replied:
I don't see how it's misleading, but if you have a better wording I'm happy to hear it.

Fikou replied:
`current job ID of the user` but it doesn't do that, it shows you ids of everyone. also `only` doesnt make sense without context of the sechud.

Fikou commented on code/modules/research/techweb/all_nodes.dm:
```diff
@@ -1139,12 +1140,13 @@
 		"nvgmesons",
 		"nv_scigoggles",
-		"security_hud_night",
+		"security_hud_night",	
```
```suggestion
		"security_hud_night",
```

Fikou commented on code/datums/components/crafting/tailoring.dm:
```diff
@@ -50,6 +50,24 @@
+/datum/crafting_recipe/hudsunciv
+	name = "Civilian HUDsunglasses"
+	result = /obj/item/clothing/glasses/hud/civ/sunglasses
+	time = 20
```
use SECONDS defines

SingingSpock replied:
I'm not quite sure what you mean by this. I more or less just copy/pasted the security HUDglasses recipe with the names/items changed.

Fikou replied:
we have defines for time. use 2 SECONDS instead of 20

Fikou commented on code/datums/components/crafting/tailoring.dm:
```diff
@@ -50,6 +50,24 @@
+	result = /obj/item/clothing/glasses/hud/civ/sunglasses
+	time = 20
+	tool_behaviors = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER)
+	reqs = list(/obj/item/clothing/glasses/hud/civ = 1,
```
format lists like this
list(
	thing1,
	thing2,
)

SingingSpock replied:
Again, this is a slightly altered copy/paste of code elsewhere in this same file. I could change it, but at that point I might as well change the rest of them for consistency and then I'm cleaning up lots of things.

Fikou replied:
ok then do that

Mothblocks replied:
you dont need to change it everywhere but its definitely good practice to change it here

SingingSpock committed:
Rewording

13spacemen commented:
Good item but isn't this called the skills HUD on Paradise? I feel like that may be a better name than civilian HUD

SingingSpock commented:
> Good item but isn't this called the skills HUD on Paradise? I feel like that may be a better name than civilian HUD

It is, but it's called the Civilian HUD on virgo codebases, and I think that name makes more sense than Skills hud. I'm not sure what the word Skills would even be referring to in that context.

SingingSpock committed:
Redoes lists and times

Redoes lists and times to fit standards. Did it for the rest of the hudglasses recipes because why not.

SingingSpock commented:
Redid the lists and times, and did the other HUDsunglasses for good measure. I'll probably go through crafting code and touch up the rest of them for GBP sometime in the nearish future if they're not up to the current standard

13spacemen commented:
yeah skills is probably not that great i feel like civilian HUD just doesn't make tell you a lot and sounds too technical for what it does (show jobs). maybe job HUD or crew HUD or work HUD or role HUD? just some ideas. good PR anyway though

GuillaumePrata commented:
>i feel like civilian HUD just doesn't tell you a lot and sounds too technical for what it does (show jobs). maybe job HUD or crew HUD or work HUD or role HUD?

Personnel HuD would be my suggestion as it is a HoP item by default and it is a job people link with checking/changing other's ID/job.
ID HuD could work too.

ATH1909 commented:
this kind of screws over people who try to bluff into places with just an outfit that makes it look like they belong there, no?

Iamgoofball commented:
``this kind of screws over people who try to bluff into places with just an outfit that makes it look like they belong there, no? and IDless antags.``
those people were screwed by a shift-click

OrionTheFox commented:
so why should it be made EASIER to screw those people over as generic crew?
moreover, there's occasions where the ID wont have a hud even if its meant to be displaying, especially with disguises, or it'll have the wrong hud.

I honestly just dont see a point to this, if you need to know what job someone is then you should be asking them, otherwise you're probably not even in a situation where you need to know. All this will do is lead to crew casually calling out anyone with no hud

SingingSpock commented:
If your ability to bluff your way into places depends on people not shift-clicking to glance at your ID, that's a skill issue. Also note that unless someone decides to get an implant (which, as has been noted, they could also get a sechud implant from the same place), this will be competing with every other form of eyewear such as mesons and medHUDs. As for the name, I'm somewhat flexible on it, but civilian feels right to me, since that's effectively the targeted audience of it. Command members can usually ask for secHUDs easily enough, it's not intended for them as much.

Comxy commented:
What about a special captains sunglasses HUD with medical and security huds combined for example

JohnFulpWillard commented on code/modules/research/designs/misc_designs.dm:
```diff
@@ -3,6 +3,24 @@
 /////////////////HUDs////////////////////
 /////////////////////////////////////////
 
+/datum/design/civ_hud
```
Shouldn't this be limited to Service since that's what it's intended for?

JohnFulpWillard reviewed (commented):
Just an idea, maybe this might fit the Lawyer as well. Would give them less of a reason to steal sechuds from security (like hop/cap).

The glasses also seem to be missing a worn icon state, and though its up to the spritetainers, I find blue to be an odd color, maybe green to differentiate from other huds while also being service-colored?

SingingSpock commented:
> What about a special captains sunglasses HUD with medical and security huds combined for example

If the captain wants more than one HUD he can get implants to do that.

JohnFulpWillard commented:
Marking as draft because my review comment, Fikou's review, and missing icon state for the glasses.

Mothblocks commented:
We have too many open PRs and this has gone stale, contact a maintainer if you can solve the conflicts and reviews
